UWP/UWPHelpers/StorageManager.cpp

```diff
@@ -0,0 +1,384 @@
+				}
+			}
+		}
+	}
+	return !subRoot.empty();
+}
+bool IsRootForAccessibleItems(std::string path)
+{
+	std::list<std::string> tmp;
+	return IsRootForAccessibleItems(Path(path), tmp, true);
+}
+
+bool GetFakeFolders(Path path, std::vector<File::FileInfo>* files, const char* filter, std::set<std::string> filters) {
+	bool state = false;
+	std::list<std::string> subRoot;
+	if (IsRootForAccessibleItems(path, subRoot)) {
+		if (!subRoot.empty()) {
+			for each (auto sItem in subRoot) {
+				auto folderPath = Path(sItem);
+				File::FileInfo info;
+				info.name = folderPath.GetFilename();
+				info.fullName = folderPath;
+				info.exists = true;
+				info.size = 1;
+				info.isDirectory = true;
+				info.isWritable = 0;
+				info.atime = 1000;
+				info.mtime = 1000;
+				info.ctime = 1000;
+				info.access = 0111;
+
+				files->push_back(info);
+				state = true;
+			}
+		}
+	}
+	return state;
+}
+
+#pragma endregion
+
+#pragma region Helpers
+bool OpenFile(std::string path) {
+	bool state = false;
+	Platform::String^ wString = ref new Platform::String(Path(path).ToWString().c_str());
+
+	StorageFile^ storageItem;
+	ExecuteTask(storageItem, StorageFile::GetFileFromPathAsync(wString));
+	if (storageItem != nullptr) {
+		ExecuteTask(state, Windows::System::Launcher::LaunchFileAsync(storageItem), false);
+	}
+	else {
+		auto uri = ref new Windows::Foundation::Uri(wString);
+		ExecuteTask(state, Windows::System::Launcher::LaunchUriAsync(uri), false);
+	}
+	return state;
+}
+
+bool OpenFolder(std::string path) {
+	bool state = false;
+	Path itemPath(path);
+	Platform::String^ wString = ref new Platform::String(itemPath.ToWString().c_str());
+	StorageFolder^ storageItem;
+	ExecuteTask(storageItem, StorageFolder::GetFolderFromPathAsync(wString));
+	if (storageItem != nullptr) {
+		ExecuteTask(state, Windows::System::Launcher::LaunchFolderAsync(storageItem), false);
+	}
+	else {
+		// Try as it's file
+		Path parent = Path(itemPath.GetDirectory());
+		Platform::String^ wParentString = ref new Platform::String(parent.ToWString().c_str());
+
+		ExecuteTask(storageItem, StorageFolder::GetFolderFromPathAsync(wParentString));
+		if (storageItem != nullptr) {
+			ExecuteTask(state, Windows::System::Launcher::LaunchFolderAsync(storageItem), false);
+		}
+	}
+	return state;
+}
+
+int64_t GetLocalFreeSpace() {
+	Platform::String^ freeSpaceKey = ref new Platform::String(L"System.FreeSpace");
+	Platform::Collections::Vector<Platform::String^>^ propertiesToRetrieve = ref new Platform::Collections::Vector<Platform::String^>();
+	propertiesToRetrieve->Append(freeSpaceKey);
+	Windows::Foundation::Collections::IMap<Platform::String^, Platform::Object^>^ result;
+	ExecuteTask(result, ApplicationData::Current->LocalFolder->Properties->RetrievePropertiesAsync(propertiesToRetrieve));
+	int64_t remainingSize = 0;
+	if (result != nullptr && result->Size > 0) {
+		try {
+			auto it = result->First();
+			auto sizeString = it->Current->Value->ToString();
```

Comment:
I'm not sure if [RetrievePropertiesAsync](https://learn.microsoft.com/en-us/uwp/api/windows.storage.fileproperties.storageitemcontentproperties.retrievepropertiesasync?view=winrt-22621) works perfectly, very little information about it,
based on docs [System.FreeSpace](https://learn.microsoft.com/en-us/windows/win32/properties/props-system-freespace) should return: 

> The amount of free space in a volume, in bytes

when I tried to cast this value (`Platform::Object`) to `int64_t` (`safe_cast<int64_t>(it->Current->Value)`)
it throwing WinRT exception `0x80004002 No such interface supported`,  
so I used `ToString()` instead, I guess this is not correct
it always report "2GB" maybe because of `ToString()`, 
it need someone with better experience to check the conversion

If we can get this to work, it will help for other drives too.

Reply:
Hm, I'm not sure.  A quick search shows people using something more akin to `result["System.FreeSpace"]` in C#/etc.  What if you set a breakpoint or log the FullName of GetType() on the object?

But I guess if we can count on it returning `#(.##)?[PTGMK]B` we can parse it worst case... although the resolution of only "2GB" isn't ideal.

-[Unknown]

Reply:
solved, I fetched the result in wrong way, it should be like
```cpp
auto value = result->Lookup(L"System.FreeSpace");
remainingSize = (uint64_t)value;
```